src/App.tsx

```diff
@@ -1,26 +1,170 @@
-/* eslint-disable max-len */
+/* eslint-disable jsx-a11y/label-has-associated-control */
 /* eslint-disable jsx-a11y/control-has-associated-label */
-import React from 'react';
-import { UserWarning } from './UserWarning';
+import React, { useEffect, useMemo, useRef, useState } from 'react';
+import * as todoService from './api/todos';
+import { Todo } from './types/Todo';
 
-const USER_ID = 0;
+import { Header } from './components/Header/Header';
+import { Footer } from './components/Footer/Footer';
+import { UserWarning } from './UserWarning';
+import { Errors } from './types/Errors';
+import { ErrorNotification } from './components/ErrorNotification';
+import { FilterType } from './types/FilterType';
+import { TodoList } from './components/TodoList/TodoList';
 
 export const App: React.FC = () => {
-  if (!USER_ID) {
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [errorMessage, setErrorMessage] = useState<Errors>(Errors.Empty);
+  const [filteredField, setFilteredField] = useState<FilterType>(
+    FilterType.All,
+  );
+  const [tempTodo, setTempTodo] = useState<Todo | null>(null);
+  const [loadingTodoIds, setLoadingTodoIds] = useState<number[]>([]);
+  const inputAddRef = useRef<HTMLInputElement>(null);
+
+  const todosActiveNumber = useMemo(() => {
+    return todos.filter(todo => !todo.completed).length;
+  }, [todos]);
+
+  const areAllTodosCompleted = useMemo(() => {
+    return todos.every(todo => todo.completed === true);
+  }, [todos]);
+
+  useEffect(() => {
+    (async () => {
+      try {
+        const data = await todoService.getTodos();
+
+        setTodos(data);
+      } catch (err) {
+        setErrorMessage(Errors.UnableToLoad);
+      }
+    })();
```

Comment:
Looks better, what do u think?

```suggestion
    const getAllTodos = async () => {
      try {
        const data = await todoService.getTodos();

        setTodos(data);
      } catch (err) {
        setErrorMessage(Errors.UnableToLoad);
      }
    }
    
    getAllTodos();
```